pkg/storage/s3/storage.go

```diff
@@ -35,6 +35,24 @@ const (
 	RWLOCK = "RWLock"
 )
 
+type multiPartUploads struct {
```

Comment:
Did you need this struct at all? Couldn't you have just used a sync.Map directly? It is nice to have the apis, but still.